native/dlapll.go

```diff
@@ -0,0 +1,37 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package native
+
+import "github.com/gonum/blas/blas64"
+
+// Dlapll returns the smallest singular value of the n×2 matrix a = [ x y ].
+// The function first computes the QR factorization of a = q*r, and then computes
+// the SVD of the 2-by-2 upper triangular matrix r.
+//
+// The contents of x and y are overwritten during the call.
+//
+// Dlapll is an internal routine. It is exported for testing purposes.
+func (impl Implementation) Dlapll(n int, x []float64, incX int, y []float64, incY int) float64 {
+	checkVector(n, x, incX)
+	checkVector(n, y, incY)
+
+	if n <= 1 {
+		return 0
+	}
+
+	// Compute the QR factorization of the N-by-2 matrix [ X Y ].
+	_, tau := impl.Dlarfg(n, x[0], x[incX:], incX)
```

Comment:
And incX != 0, so I don't need to reset x[0] to 1, since x[0:] cannot be a parameter. True?